framework/Optimizers/survivorSelectors/survivorSelectors.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+  Implementation of survivorSelctors (Elitism) for new generation
+  selection process of Genetic Algorithm. Currently the implemented
+  survivorSelctors algorithms are:
+  1.  ageBased
+  2.  fitnessBased
+
+  Created June,16,2020
+  @authors: Mohammad Abdo, Diego Mandelli, Andrea Alfonsi
+"""
+
+import numpy as np
+import xarray as xr
+import copy
+# @profile
+def ageBased(newRlz,**kwargs):
+  """
+    ageBased survivorSelection mechanism for new generation selection.
+    It replaces the oldest parents with the new children regardless of the fitness.
+    @ In, newRlz, xr.DataSet, containing either a single realization, or a batch of realizations.
+    @ In, kwargs, dict, dictionary of parameters for this mutation method:
+          age, list, age list for each chromosome of the previous population
+          variables, list of variable names to be sampled
+          fitness, xr.DataArrays, fitness of the previous generation
+          offSpringsFitness, xr.DataArray, fitness of each new child, i.e., np.shape(offSpringsFitness) = nChildren x nGenes
+          population, xr.DataArray, population from previous generation
+    @ Out, newPopulation, xr.DataArray, newPopulation for the new generation, i.e. np.shape(newPopulation) = populationSize x nGenes.
+    @ Out, newFitness, xr.DataArray, fitness of the new population
+    @ Out, newAge, list, Ages of each chromosome in the new population.
```

Comment:
Hm, is there a way to specify something like remove the 10 oldest, but keep the most fit one? (Since otherwise you might remove the best one found.)  This doesn't need to be added now, but if it is not possible it would be useful to have some way of always keeping the most fit (or 5 most fit...).